tree/ntuple/v7/src/RColumn.cxx

```diff
@@ -61,10 +66,22 @@ void ROOT::Experimental::Detail::RColumn::Connect(DescriptorId_t fieldId, RPageS
 
 void ROOT::Experimental::Detail::RColumn::Flush()
 {
-   if (fHeadPage.GetSize() == 0) return;
+   auto otherIdx = 1 - fHeadPageIdx;
+   if (fHeadPage[fHeadPageIdx].IsEmpty() && fHeadPage[otherIdx].IsEmpty())
+      return;
 
-   fPageSink->CommitPage(fHandleSink, fHeadPage);
-   fHeadPage.Reset(fNElements);
+   if ((fHeadPage[fHeadPageIdx].GetNElements() < fApproxNElementsPerPage / 2) && !fHeadPage[otherIdx].IsEmpty()) {
+      // Small tail page: merge with previously used page; we know that there is enough space in the shadow page
+      void *dst = fHeadPage[otherIdx].GrowUnchecked(fHeadPage[fHeadPageIdx].GetNElements());
+      RColumnElementBase elem(fHeadPage[fHeadPageIdx].GetBuffer(), fHeadPage[fHeadPageIdx].GetElementSize());
+      elem.WriteTo(dst, fHeadPage[fHeadPageIdx].GetNElements());
+      fHeadPage[fHeadPageIdx].Reset(0);
+      std::swap(fHeadPageIdx, otherIdx);
```

Comment:
The only use of the swap (vs a straight assignment) seems to be being able to assert that the 'other' page is empty on line 82, is that worth it?  (and if the assert was `R__DEBUG_ASSERT` or (`std::assert`) wouldn't be better to just redo the `1 - ` operation inside the assert?

Reply:
I thought it is worth it because the alternative would be two assert statements, one if the `if` statement is taken and one otherwise. The extra cycles should be negligible compared to the `fPageSink->CommitPage` (<buffered> write) call.  But I don't feel strongly about it, let me know if you still prefer the code to be changed.

Reply:
I feel strongly that we should remove the swap  *** if and only if *** we choose to have the assert changed to be on "only in debug build"

> be two assert statements ..

humm actually I thought it would be only one:

```
R__ASSERT(fHeadPage[1 - fHeadPageIdx].IsEmpty());
```